### PR TITLE
Try to reproduce Khan et al 2015

### PR DESCRIPTION
**Pull request by jasmainak**
I was trying to reproduce [Khan et al 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5013931/pdf/awv043.pdf) with the new API. Really happy about the interface for the most part. It took me only 5 minutes to set it up and reproduce the paper. Couple of comments:

- spike ISI is not exposed in the API and numspikes can't be 3. This is a known issue but it's a good motivation to fix it
- Maybe there should be reasonable defaults for `weights_nmda` and `weights_ampa` ? I copied it this time but having it as a default might be even better
- this script produces an error. Some implicit assumptions in the plotting functions @cjayb 

cc @SherazKhan

SSVEP

### CODE REVIEW

File: examples/plot_simulate_khan.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from hnn_core import simulate_dipole, read_params, Network, read_spikes
+from hnn_core.viz import plot_dipole
+
+hnn_core_root = op.dirname(hnn_core.__file__)
+
+###############################################################################
+# Then we read the parameters file
+params_fname = op.join(hnn_core_root, 'param', 'default.json')
+params = read_params(params_fname)
+params['tstop'] = 1000
```
Comment (jasmainak) [inline examples/plot_simulate_khan.py:26]: change to something small for faster debugging

**Comment by cjayb** [comment]
COOL! Those should definitely be exposed. If nothing else, for our sanity so we don't have arbitrary +/-5's in there :)

**Comment by cjayb** [comment]
My reading of the Khan and Jones papers yields:

```python
weights_ampa = {'L2_basket': 0.006562, 'L2_pyramidal': .000007,
                'L5_pyramidal': 0.142300}
weights_nmda = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
                'L5_pyramidal': 0.080074}
t0 = 0
numspikes = 2
spike_isi = 7
repeats = 10
location = 'proximal'
net.add_bursty_drive(
        'bursty1', distribution='normal', t0=t0, sigma_t0=2.,
        T=params['tstop'], burst_f=25, burst_sigma_f=2.,
        numspikes=numspikes, spike_isi=spike_isi, repeats=repeats,
        weights_ampa=weights_ampa, weights_nmda=weights_nmda,
        location=location, seedcore=2, space_constant=100.)

t0 = 14
# XXX: original paper says 3, also 7 ms instead of 10 ms ISI
numspikes = 3
spike_isi = 7
location = 'distal'
net.add_bursty_drive(
        'bursty2', distribution='normal', t0=t0, sigma_t0=2.,
        T=params['tstop'], burst_f=25, burst_sigma_f=2., numspikes=numspikes,
        repeats=repeats, weights_ampa=weights_ampa, weights_nmda=weights_nmda,
        location=location, seedcore=2, spike_isi=spike_isi,
        space_constant=100.)
```

The result doesn't look plausible. The 2009 paper isn't much help in figuring things out: I could not find the FF/FB weights anywhere (where did you get them from?), and the variance of the spikes around each burst center (`burst_sigma_f`, poorly named btw) is a bit of a mystery to me.

Screenshot 2020-12-30 at 15 15 52 Screenshot 2020-12-30 at 15 16 00

**Comment by jasmainak** [comment]
why do you say it doesn't look plausible? couple of points:

- it's the 2015 paper, not 2009 ... not sure which one you're referring to?
- I wonder if we should consider renaming the parameters to be "burst_freq" and "burst_freq_sigma". But let's not try to do everything in one PR. I suggest that we try to converge a bit.
- The paper says "The simulations ran for 1 s and the first 550 ms were discarded to allow for initialization of the model’s internal dynamics." This seems consistent with what we see in the spiking pattern of L5 Pyramidal cells above?
- For the weights I just copied the weights from the default evoked simulation

**Comment by cjayb** [comment]
> * For the weights I just copied the weights from the default evoked simulation

But there aren't any bursty ("common") inputs applied in the evoked sim, are there? So the weights might be totally bogus. I skimmed both Khan 2015 and Jones 2009 in the hopes of finding some values, but didn't. I don't like the way those spike raster plots look, and the dipole moment looks almost sinusoidal. Makes me think the weights are too high.

**Pull request closed without merging**